### ISSUE TITLE
Postgres based TimescalDB back-up difficulties

### ISSUE THREAD

**Issue by Flight777** [tiredofit/docker-db-backup#54]
First of all huge thanks for making this tool available and help to get rid of the back-up hassle!

Currently I'm trying to use it for backing up a TimescaleDB but I'm encountering some issues, so hopefully you are able to help me with:

```
[NOTICE] ** [db-backup] Compressing backup with gzip


pg_dump: warning: there are circular foreign-key constraints on this table:


pg_dump:   hypertable


pg_dump: You might not be able to restore the dump without using --disable-triggers or temporarily dropping the constraints.


pg_dump: Consider using a full dump instead of a --data-only dump to avoid this problem.


pg_dump: warning: there are circular foreign-key constraints on this table:


pg_dump:   chunk


pg_dump: You might not be able to restore the dump without using --disable-triggers or temporarily dropping the constraints.


pg_dump: Consider using a full dump instead of a --data-only dump to avoid this problem.


[NOTICE] ** [db-backup] Generating MD5 for pgsql_exsyn_TimescaleDB_20210121-114348.sql.gz


[NOTICE] ** [db-backup] Backup of pgsql_exsyn_TimescaleDB_20210121-114348.sql.gz created with the size of 1572 bytes


[INFO] ** [db-backup] Dumping database: decide


[NOTICE] ** [db-backup] Compressing backup with gzip


pg_dump: warning: there are circular foreign-key constraints on this table:


pg_dump:   hypertable


pg_dump: You might not be able to restore the dump without using --disable-triggers or temporarily dropping the constraints.


pg_dump: Consider using a full dump instead of a --data-only dump to avoid this problem.


pg_dump: warning: there are circular foreign-key constraints on this table:


pg_dump:   chunk


pg_dump: You might not be able to restore the dump without using --disable-triggers or temporarily dropping the constraints.


pg_dump: Consider using a full dump instead of a --data-only dump to avoid this problem.


pg_dump: NOTICE:  hypertable data are in the chunks, no data will be copied


DETAIL:  Data for hypertables are stored in the chunks of a hypertable so COPY TO of a hypertable will not copy any data.


HINT:  Use "COPY (SELECT * FROM <hypertable>) TO ..." to copy all data in hypertable, or copy each chunk individually.


pg_dump: NOTICE:  hypertable data are in the chunks, no data will be copied


DETAIL:  Data for hypertables are stored in the chunks of a hypertable so COPY TO of a hypertable will not copy any data.


HINT:  Use "COPY (SELECT * FROM <hypertable>) TO ..." to copy all data in hypertable, or copy each chunk individually.


pg_dump: NOTICE:  hypertable data are in the chunks, no data will be copied


DETAIL:  Data for hypertables are stored in the chunks of a hypertable so COPY TO of a hypertable will not copy any data.


HINT:  Use "COPY (SELECT * FROM <hypertable>) TO ..." to copy all data in hypertable, or copy each chunk individually.


pg_dump: NOTICE:  hypertable data are in the chunks, no data will be copied


DETAIL:  Data for hypertables are stored in the chunks of a hypertable so COPY TO of a hypertable will not copy any data.


HINT:  Use "COPY (SELECT * FROM <hypertable>) TO ..." to copy all data in hypertable, or copy each chunk individually.


pg_dump: NOTICE:  hypertable data are in the chunks, no data will be copied


DETAIL:  Data for hypertables are stored in the chunks of a hypertable so COPY TO of a hypertable will not copy any data.


HINT:  Use "COPY (SELECT * FROM <hypertable>) TO ..." to copy all data in hypertable, or copy each chunk individually.


[NOTICE] ** [db-backup] Generating MD5 for pgsql_decide_TimescaleDB_20210121-114348.sql.gz


[NOTICE] ** [db-backup] Backup of pgsql_decide_TimescaleDB_20210121-114348.sql.gz created with the size of 229055344 bytes


[NOTICE] ** [db-backup] Sending Backup Statistics to Zabbix
```

Is there a possibility to force a full pgsql dump to circumvent this? 

Native backup support for TimescaleDB would even be greater of course, but I'm not sure how to implement their guidelines with your tool: https://docs.timescale.com/latest/using-timescaledb/backup

Please let me know if you need any additional info to provide a meaningful answer.

Thanks,

Rob

**Comment by echoblag** [comment]
@Flight777 I have run into this and done some digging.

I need to investigate this further because regardless of these warnings the database backup job hasn't completed for the database, so I don't think they're as harmless as claimed.

Let me know if you have made any progress.

TL;DR Timescale DB says to ignore them.
- https://github.com/timescale/timescaledb/issues/1581
- https://docs.timescale.com/self-hosted/latest/troubleshooting/#versions-are-mismatched-when-dumping-and-restoring-a-database
